### PR TITLE
Adding support for redis logical db

### PR DESCRIPTION
**Pull request by 999lakhisidhu**
### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

File: docs/usage/configuration/config_documentation.md [hunk 1]
```diff
@@ -3927,6 +3927,8 @@ This setting has the following sub-options:
 * `host` and `port`: Optional host and port to use to connect to redis. Defaults to
    localhost and 6379
 * `password`: Optional password if configured on the Redis instance.
+* `dbid`: Optional redis dbid if needs to connect to specific redis logical db.
```
Comment (DMRobertson) [inline docs/usage/configuration/config_documentation.md:3930]: Can you include a note that this was added in Synapse 1.78? See https://github.com/matrix-org/synapse/blob/1617f7881df51e8cc9fbf0d30c098a6b586138b7/docs/usage/configuration/config_documentation.md?plain=1#L445-L452 for an example.

**Review by DMRobertson — DISMISSED** [review]
As well as the changelog note, have you have a chance to test this?

**Comment by 999lakhisidhu** [comment]
@DMRobertson  I have incorporated  your comments .. and I have tested this feature on my synapse environment.

File: docs/usage/configuration/config_documentation.md [hunk 2]
```diff
@@ -3927,6 +3927,8 @@ This setting has the following sub-options:
...
+   _Added in Synapse 1.78.0
```
Comment (DMRobertson) [inline docs/usage/configuration/config_documentation.md:3931]:
```suggestion

  _Added in Synapse 1.78.0._
```

**Review by MatMaul — APPROVED** [review]
LGTM.

**Comment by 999lakhisidhu** [comment]
@DMRobertson can we please close this PR if nothing is pending ?

**Pull request merged**